release-notes/9.0/supported-os.json

```diff
@@ -0,0 +1,304 @@
+{
+    "channel-version": "9.0",
+    "last-updated": "2024-05-01",
+    "operating-systems": [
+        {
+            "family": "Android",
+            "distributions": [
+                {
+                    "name": "Android",
+                    "lifecycle-policy": "https://support.google.com/android",
+                    "architectures": [
+                        "x64",
+                        "arm64",
+                        "arm"
+                    ],
+                    "supported-cycles": [
+                        "API 21+"
```

Comment:
While what you wrote is correct we have been using the API version instead of the Android OS version pretty much everywhere else, and it it's also what a user would set for `SupportedOSPlatformVersion` in the project file.

I'd prefer if we use API version for consistency.

Reply:
My main issue is what I wrote here: https://github.com/dotnet/core/pull/9294#pullrequestreview-2037531038

The current `supported-os.md` is mixing up `SupportedOSPlatformVersion` and `supported-cycles`. `SupportedOSPlatformVersion` for Windows is not even mentioned anywhere.

For me as an end-user it's pretty hard to keep everything locally in sync because our end-customers are not interested in our `SupportedOSPlatformVersion` but about what OS we support them on. And I am interested on that as well from the perspective for what OS I get support from Microsoft if I stumble into issues. The documentation is currently only covering half of that and also not very consistent.

Reply:
Yes I agree with that. We've been struggling to clearly define/separate
1. the minimum version an app will run on constrained by technical means i.e. min-sdk, min libc etc.
2. the minimum version we "officially" support which excludes EOL versions

Reply:
https://github.com/dotnet/core/issues/8880 can give you a better picture about where this thing started of from.

Reply:
In my opinion, the information about OS EOLs should be maintained separatedly and since the JSON files are generated according to @leecow should be taken into account when generating these to not manually maintain them in multiple locations. Otherwise the `supported-cycles` would need to be kept in sync for all .NET channels on each EOL.

Reply:
The most important bit of data is supported OS, which aligns with "what we test and support". We should make sure that's captured. We've run into confusion in the past when we didn't have clarity on this.

What we target (with the compiler) is also important.

Reply:
It seems like https://github.com/dotnet/core/pull/9294#discussion_r1588869286 is a compromise that gives everyone what they want. I'm not seeing an obvious downside to this approach. Yes/No?